### PR TITLE
fix docs error in Autograd Mechanics

### LINKED ISSUE #74682: doc issue in AUTOGRAD MECHANICS
### 📚 The doc issue

In the section **Evaluation Mode** (`nn.Module.eval()`)

https://pytorch.org/docs/stable/notes/autograd.html#evaluation-mode-nn-module-eval

> Functionally, `module.eval()` (or equivalently `module.train()`) are completely orthogonal to no-grad mode and inference mode.

Should be `module.train(False)`

### Suggest a potential alternative/fix

Functionally, `module.eval()` (or equivalently `module.train(False)`) are completely orthogonal to no-grad mode and inference mode.

cc @brianjo @mruberry

### PR DESCRIPTION
**Pull request by bhbbbbb**
Fixes #74682

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Thanks for the fix
